### PR TITLE
Certificate update: add stake key registration, delegation and get-pool-id

### PR DESCRIPTION
**Pull request by NicolasDP**

### CODE REVIEW

File: src/bin/jcli_app/certificate/mod.rs [hunk 1]
```diff
@@ -1,36 +1,41 @@
...
-    StakePoolRegistration(StakePoolRegistrationArgs),
-}
-
-#[derive(StructOpt)]
-pub struct StakeKeyRegistrationArgs {
-    #[structopt(name = "PUBLIC_KEY")]
-    pub key: String,
-    #[structopt(name = "SIGNING_KEY")]
-    pub private_key: PathBuf,
+    /// build a stake poole registration certificate
```
Comment (mzabaluev) [inline src/bin/jcli_app/certificate/mod.rs]: Typo: pool

File: utils/src/certificate.rs [hunk 2]
```diff
@@ -2,53 +2,13 @@ use bech32::{Bech32, FromBase32, ToBase32};
...
-    },
-    ReadError(ReadError),
-}
-
-impl From<bech32::Error> for Error {
-    fn from(error: bech32::Error) -> Self {
-        Error::Bech32Error(error)
-    }
-}
+type StaticStr = &'static str;
```
Comment (mzabaluev) [inline utils/src/certificate.rs]: Eh, why obscure the common built-in type signature with a typedef?

Comment (NicolasDP) [reply utils/src/certificate.rs]: the `custom_error` macro does not like the raw version, this is a bit cir-convoluted way to go around the limitations.

File: src/bin/jcli_app/certificate/mod.rs (hunk 1 above)
Comment (NicolasDP) [reply src/bin/jcli_app/certificate/mod.rs]: cheers

**Pull request merged**